packages/postcss-lowercase-text/src/__tests__/atRules.test.js

```diff
@@ -0,0 +1,120 @@
+  );
+});
+
+describe('transforming @namespace', () => {
+  run(
+    'should safely transform the @namespace',
+    '@NAMESPACE prefix url(http://www.w3.org/1999/xhtml);',
+    '@namespace prefix url(http://www.w3.org/1999/xhtml);'
+  );
+});
+
+describe('transforming @supports ', () => {
+  run(
+    'should safely transform the @supports ',
+    '@SUPPORTS (display: grid) {div {display: grid;}};@supports not (display: GRID) {div {float: right;}}',
+    '@supports (display: grid) {div {display: grid;}};@supports not (display: GRID) {div {float: right;}}'
+  );
+});
+
+describe('transforming @document  ', () => {
+  run(
+    'should safely transform the @document  ',
+    '@DOCUMENT URL("https://www.EXAMPLE.com/") {h1 {COLOR: green;}}',
+    '@document URL("https://www.EXAMPLE.com/") {h1 {color: green;}}'
+  );
+});
+
+describe('transforming @page   ', () => {
+  run(
+    'should safely transform the @page   ',
+    '@PAGE {margin: 1cm;};@page :FIRST {margin: 2cm;}',
+    '@page {margin: 1cm;};@page :FIRST {margin: 2cm;}'
+  );
+});
+
+describe('transforming @font-face', () => {
+  run(
+    'should safely transform the @font-face   ',
+    '@FONT-FACE {font-family: "Open Sans";src: url("/fonts/OpenSans-Regular-webfont.woff2") format("woff2"),url("/fonts/OpenSans-Regular-webfont.woff") FORMAT("woff");}',
+    '@font-face {font-family: "Open Sans";src: url("/fonts/OpenSans-Regular-webfont.woff2") format("woff2"),url("/fonts/OpenSans-Regular-webfont.woff") FORMAT("woff");}'
+  );
+});
+
+describe('transforming @keyframes', () => {
+  run(
+    'should safely transform the @keyframes',
+    '@KEYFRAMES slidein {from {transform: translateX(0%);}};@KEYFRAMES important2  {FROM {TRANSFORM: translateX(0%);}}',
+    '@keyframes slidein {from {transform: translateX(0%);}};@keyframes important2  {from {transform: translateX(0%);}}'
+  );
+});
+
+describe('transforming @viewports', () => {
+  run(
+    'should safely transform the @viewports',
+    '@VIEWPORT {min-width: 640px;max-width: 800px;}@viewport {ZOOM: 0.75;  min-zoom: 0.5;max-zoom: 0.9;}@viewport {orientation: LANDSCAPE;}',
+    '@viewport {min-width: 640px;max-width: 800px;}@viewport {zoom: 0.75;  min-zoom: 0.5;max-zoom: 0.9;}@viewport {orientation: LANDSCAPE;}'
+  );
+});
+
+describe('transforming @counter-style', () => {
+  run(
+    'should safely transform the @counter-style',
+    '@counter-STYLE thumbs {system: cyclic;SYMBOLs: "F44D";suffix: " ";} UL{list-style: thumbs;}',
+    '@counter-style thumbs {system: cyclic;symbols: "F44D";suffix: " ";} ul{list-style: thumbs;}'
+  );
+});
+
+describe('transforming @counter-style', () => {
+  run(
+    'should safely transform the @counter-style',
+    '@counter-STYLE CAPS {system: cyclic;SYMBOLs: "F44D";suffix: " ";} UL{list-style: CAPS;} .ulstyle {list-style: caps;}',
+    '@counter-style caps {system: cyclic;symbols: "F44D";suffix: " ";} ul{list-style: CAPS;} .ulstyle {list-style: caps;}'
```

Comment:
CAPS should not be lowercased, it is ident

Reply:
>  it is ident

I didnt get this

`@counter-style` is case insensitive I guess so its safe to transform

Reply:
```
@counter-style <counter-style-name> {
  [ system: <counter-system>; ] ||
  [ symbols: <counter-symbols>; ] ||
  [ additive-symbols: <additive-symbols>; ] ||
  [ negative: <negative-symbol>; ] ||
  [ prefix: <prefix>; ] ||
  [ suffix: <suffix>; ] ||
  [ range: <range>; ] ||
  [ pad: <padding>; ] ||
  [ speak-as: <speak-as>; ] ||
  [ fallback: <counter-style-name>; ]
}
where 
<counter-style-name> = <custom-ident>
```

ident never should be lowercased

Reply:
ohh okok.... THanks for pointing.